tty/grlib-uart/grlib-uart.c

```diff
@@ -83,18 +90,31 @@ typedef struct {
 static const struct {
 	uint32_t *base;
 	unsigned int irq;
-	unsigned int cgudev;
 	uint8_t txPin;
 	uint8_t rxPin;
 } info[UART_MAX_CNT] = {
-	{ UART0_BASE, UART0_IRQ, cgudev_apbuart0, UART0_TX, UART0_RX },
-	{ UART1_BASE, UART1_IRQ, cgudev_apbuart1, UART1_TX, UART1_RX },
-	{ UART2_BASE, UART2_IRQ, cgudev_apbuart2, UART2_TX, UART2_RX },
-	{ UART3_BASE, UART3_IRQ, cgudev_apbuart3, UART3_TX, UART3_RX },
-	{ UART4_BASE, UART4_IRQ, cgudev_apbuart4, UART4_TX, UART4_RX },
-	{ UART5_BASE, UART5_IRQ, cgudev_apbuart5, UART5_TX, UART5_RX }
+	{ UART0_BASE, UART0_IRQ, UART0_TX, UART0_RX },
+	{ UART1_BASE, UART1_IRQ, UART1_TX, UART1_RX },
+	{ UART2_BASE, UART2_IRQ, UART2_TX, UART2_RX },
+	{ UART3_BASE, UART3_IRQ, UART3_TX, UART3_RX },
+	{ UART4_BASE, UART4_IRQ, UART4_TX, UART4_RX },
+	{ UART5_BASE, UART5_IRQ, UART5_TX, UART5_RX }
+};
+
+#if defined(__CPU_GR716)
+
+static const struct {
+	unsigned int dev;
+} cguInfo[UART_MAX_CNT] = {
```

Comment:
A way to hide this inside of uart_cguInit - pass index instead of cgudev_apbuart*